airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -224,10 +225,11 @@ def _filter_dynamic_fields(self, records: Iterable) -> Iterable:
         see https://github.com/airbytehq/airbyte/issues/2397
         """
         for record in records:
-            if isinstance(record, Mapping) and "properties" in record:
-                for key in list(record["properties"].keys()):
-                    if key.startswith("hs_time_in"):
-                        record["properties"].pop(key)
+            if self.unset_dynamic_fields:
+                if isinstance(record, Mapping) and "properties" in record:
+                    for key in record["properties"]:
+                        if key.startswith("hs_time_in_"):
+                            record["properties"][key] = None
```

Comment:
Could you move some logic to another method? It's so many nestings and difficult to read this

Reply:
What do you think about new approach? At least it has to increase performance.

Reply:
Looks great!